src/components/Arrow.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const BaseArrow = styled.div`
+  position: absolute;
+  width: 0;
+  height: 0;
+  content: '';
+`;
+
+const ArrowDown = styled(BaseArrow)`
+  bottom: -8px;
+  border-left: 12px solid transparent;
+  border-right: 12px solid transparent;
+  border-top: 12px solid black;
+`;
+
+const ArrowUp = styled(BaseArrow)`
+  top: -8px;
+  border-left: 12px solid transparent;
+  border-right: 12px solid transparent;
+  border-bottom: 12px solid black;
+`;
+
+const ArrowLeft = styled(BaseArrow)`
+  left: -8px;
+  border-top: 12px solid transparent;
+  border-bottom: 12px solid transparent;
+  border-right: 12px solid black;
+`;
+
+const ArrowRight = styled(BaseArrow)`
+  right: -8px;
+  border-top: 12px solid transparent;
+  border-bottom: 12px solid transparent;
+  border-left: 12px solid black;
+`;
+
+interface ArrowMap {
+  [key: string]: typeof BaseArrow;
```

Comment:
Maybe call this `[direction: string]` (or similar) to make the `t`, `b`, `l`, `r` initials clearer?